### PR TITLE
Added DAO Layer

### PR DESCRIPTION
**Pull request by eyouu**

### CODE REVIEW

File: core/pom.xml [hunk 1]
```diff
@@ -33,25 +33,9 @@
...
     <dependencies>
         <dependency>
-            <groupId>com.h2database</groupId>
-            <artifactId>h2</artifactId>
-            <version>1.4.200</version>
-            <scope>test</scope>
-        </dependency>
-        <dependency>
-            <groupId>org.liquibase</groupId>
-            <artifactId>liquibase-maven-plugin</artifactId>
```
Comment (antonbakalets) [inline core/pom.xml:43]: org.liquibase will be moved in a separate task

File: core/pom.xml [hunk 2]
```diff
@@ -63,11 +47,6 @@
             <scope>runtime</scope>
             <optional>true</optional>
         </dependency>
-        <dependency>
-            <groupId>org.postgresql</groupId>
-            <artifactId>postgresql</artifactId>
```
Comment (antonbakalets) [inline core/pom.xml:68]: Postgres should stay. `Core` module is deciding which database to use.

File: core/pom.xml [hunk 3]
```diff
@@ -33,25 +33,9 @@
...
-                <exclusion>
-                    <groupId>junit</groupId>
-                    <artifactId>junit</artifactId>
-                </exclusion>
-            </exclusions>
-        </dependency>
-        <dependency>
-            <groupId>org.springframework.boot</groupId>
-            <artifactId>spring-boot-starter-data-jpa</artifactId>
+            <groupId>greencity</groupId>
```
Comment (antonbakalets) [inline core/pom.xml:36]: `${project.groupId}` I believe

File: dao/pom.xml [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+        <modelmapper.version>2.3.5</modelmapper.version>
+        <jackson.version>2.10.1</jackson.version>
+        <powermock.version>2.0.4</powermock.version>
+        <mockito-core.version>3.3.3</mockito-core.version>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-web</artifactId>
```
Comment (antonbakalets) [inline dao/pom.xml:30]: No web here, only database.

File: dao/pom.xml [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+            <version>${liquibase.version}</version>
+            <exclusions>
+                <exclusion>
+                    <groupId>junit</groupId>
+                    <artifactId>junit</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>org.postgresql</groupId>
```
Comment (antonbakalets) [inline dao/pom.xml:65]: Please remove.

File: dao/pom.xml [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+            <exclusions>
+                <exclusion>
+                    <groupId>junit</groupId>
+                    <artifactId>junit</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-core</artifactId>
```
Comment (antonbakalets) [inline dao/pom.xml:82]: Tests are not moved yet.

File: dao/pom.xml [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+            <version>${mockito-core.version}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.powermock</groupId>
```
Comment (antonbakalets) [inline dao/pom.xml:87]: Please avoid powermock.

File: dao/pom.xml [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+            <artifactId>junit-jupiter-engine</artifactId>
+            <version>${junit-jupiter.version}</version>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>com.microsoft.azure</groupId>
```
Comment (antonbakalets) [inline dao/pom.xml:103]: This module is not deployed. It is included into `Core` module.

**Review by antonbakalets — CHANGES_REQUESTED** [review]

File: dao/pom.xml (hunk 4 above)
Comment (eyouu) [reply dao/pom.xml:20]: If I delete it, I won't have `javax-validation` in my entities

Comment (antonbakalets) [reply dao/pom.xml:20]: Add what you need only. Add javax-validation.

Comment (antonbakalets) [reply dao/pom.xml:20]:
```
<dependency>
      <groupId>javax.annotation</groupId>
      <artifactId>javax.annotation-api</artifactId>
      <version>1.3.2</version>
    </dependency>
```
to be presize

File: core/pom.xml [hunk 9]
```diff
@@ -49,30 +63,11 @@
...
-            <scope>runtime</scope>
-            <optional>true</optional>
-        </dependency>
         <dependency>
             <groupId>org.postgresql</groupId>
             <artifactId>postgresql</artifactId>
             <scope>runtime</scope>
         </dependency>
-        <dependency>
-            <groupId>org.projectlombok</groupId>
```
Comment (antonbakalets) [inline core/pom.xml:72]: Lombook is an optional dependency, so it's not available from dao. It should stay.

File: dao/src/main/java/greencity/DaoApplication.java [hunk 10]
```diff
@@ -0,0 +1,14 @@
+package greencity;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class DaoApplication {
```
Comment (antonbakalets) [inline dao/src/main/java/greencity/DaoApplication.java:7]: You can just remove for the moment.

**Review by antonbakalets — APPROVED** [review]

**Review by forestik — APPROVED** [review]

**Review by TarasDovganyuk — APPROVED** [review]

**Pull request merged**